archive/imo/imo2021_q1.lean

```diff
@@ -0,0 +1,299 @@
+/-
+Copyright (c) 2021 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+
+import data.real.sqrt
+import tactic.interval_cases
+import tactic.linarith
+import tactic.norm_cast
+import tactic.norm_num
+import tactic.ring_exp
+
+
+/-!
+# IMO 2021 Q1
+Let `n≥100` be an integer. Ivan writes the numbers` n, n+1,..., 2n` each on different cards.
+He then shuffles these `n+1` cards, and divides them into two piles. Prove that at least one
+of the piles contains two cards such that the sum of their numbers is a perfect square.
+# Solution
+We show there exists a triplet `a, b, c ∈ [n , 2n]` with `a < b < c` and each of the sums `(a + b)`,
```

Comment:
Blank lines around headings is our convention, even though markdown doesn't care:
```suggestion
# IMO 2021 Q1

Let `n≥100` be an integer. Ivan writes the numbers` n, n+1,..., 2n` each on different cards.
He then shuffles these `n+1` cards, and divides them into two piles. Prove that at least one
of the piles contains two cards such that the sum of their numbers is a perfect square.

# Solution

We show there exists a triplet `a, b, c ∈ [n , 2n]` with `a < b < c` and each of the sums `(a + b)`,
```